### PR TITLE
initial docs with example for pd upgrade

### PR DESCRIPTION
**Pull request by samir-gandhi**
- added an example that should walk through a successful upgrade. 
- documented the example with some thoughts for best practices. 

probably should add a diagram.. but didn't

### CODE REVIEW

**Comment by kbping** [comment]
Looks good to me, Samir. I'll do an edit pass after you post.

File: docs/upgradePingDirectory.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+> This document will focus on a PingDirectory upgrade specifically in a default Kubernetes environment. 
+
+As an example, we'll walk through upgrading a PingDirectory Statefulset 8.0.0.1 to 8.1.0.0 via an incremental canary roll-out. 
+
+## Tips
+
+To build confidence in the upgrade process, follow some considerations:
+
+* Avoid combining configuration changes and version upgrades in the same rollout. This add unnecessary complexity to debugging errors during an upgrade.
```
Comment (tsigle) [inline docs/upgradePingDirectory.md:13]: "add" should be plural

File: 20-kubernetes/12-pingdirectory-upgrade/3-staging.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      name: pingdirectory
+    spec:
+      containers:
+      - envFrom:
+        - configMapRef:
+            name: pingdirectory-environment-variables
+        - secretRef:
+            name: devops-secret
+            optional: true
+        image: pingidentity/pingdirectory:8.1.0.0
```
Comment (tsigle) [inline 20-kubernetes/12-pingdirectory-upgrade/3-staging.yaml:67]: Maybe place a comment before this that it is the important change.  Call it out to it's obvious.

File: 20-kubernetes/12-pingdirectory-upgrade/4-rollout.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+kind: StatefulSet
+metadata:
+  labels:
+    role: pingdirectory
+  name: pingdirectory
+spec:
+  replicas: 2
+  updateStrategy:
+    type: RollingUpdate
+  selector:
```
Comment (tsigle) [inline 20-kubernetes/12-pingdirectory-upgrade/4-rollout.yaml:47]: Maybe a comment that the partition is being removed.

**Comment by samir-gandhi** [comment]
updated per terry's comments. merging so @kbping can look further.

**Pull request merged**